oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -685,6 +686,9 @@ private void collectDetailedGarbage(final GCPhases phases, final RevisionVector
                                             modified, fromModifiedMs, toModifiedMs);
                                 }
                             }
+                            if (infoFullGCEnabled) {
+                                return;
```

Comment:
there's a problem here : with returning early, the part of the code after the `removeGarbage` bit, is not executed. This would mean the iterating logic would somehow break. I think it would be good to have a test case that proves it would break, and then fix this bit.

Reply:
Rather than adding the `if-else` check here, I would ignore the deletion part inside `removeGarbage()`.

It would keep the flow simple and inside the `dryRun` mode, we can `log` the `properties/documents` that would have been deleted.

The same should be followed for other deletion operations for `deletingDocs` & `splitCleanUp`.

Following this approach wouldn't skip the iterating logic and thus avoid any unintended consequences.

cc @stefan-egli @mreutegg